src/spaceObjects/spaceship.cpp

```diff
@@ -54,12 +54,28 @@ REGISTER_SCRIPT_SUBCLASS_NO_CREATE(SpaceShip, ShipTemplateBasedObject)
     REGISTER_SCRIPT_CLASS_FUNCTION(SpaceShip, setSystemPowerFactor);
     REGISTER_SCRIPT_CLASS_FUNCTION(SpaceShip, getSystemCoolant);
     REGISTER_SCRIPT_CLASS_FUNCTION(SpaceShip, setSystemCoolant);
+    ///[Deprecated] Gets forward impulse max speed.
+    ///Consider using getBothImpulseMaxSpeeds as forward and reverse speed can be different.
     REGISTER_SCRIPT_CLASS_FUNCTION(SpaceShip, getImpulseMaxSpeed);
+    ///Sets max speed.
+    ///If called with only one argument, sets forward and reverse speed to equal values.
+    ///If called with two arguments, first one is forward speed and second one is reverse speed.
     REGISTER_SCRIPT_CLASS_FUNCTION(SpaceShip, setImpulseMaxSpeed);
+    ///Get multiple results, first one is forward speed and second one is reverse speed.
+    ///ex : forward,reverse = getBothImpulseMaxSpeeds() (you can also use select or _ to get only reverse speed)
+    REGISTER_SCRIPT_CLASS_FUNCTION(SpaceShip, getBothImpulseMaxSpeeds);
     REGISTER_SCRIPT_CLASS_FUNCTION(SpaceShip, getRotationMaxSpeed);
     REGISTER_SCRIPT_CLASS_FUNCTION(SpaceShip, setRotationMaxSpeed);
+    ///[Deprecated] Gets forward impulse acceleration.
+    ///Consider using getBothAccelerations as forward and reverse acceleration can be different.
```

Comment:
I don't think this one needs to be depricated. Remembering my lua right, if you now do:
```
local acc = ship:getBothAccelerations()
```
you just get the forward acceleration, so the getAcceleration implementation can just be replaced with the getBothAccelerations implementation, and getBothAccelerations can be dropped. (yay, less code)

(same for getting speed)

Reply:
Yes I wanted to go with this :)
For me it would break this kind of usage
x,t = getImpulseMaxSpeed(),OtherFunctionReturningT() 
or 
string.format("%f %f %f",getImpulseMaxSpeed(), OtherFunctionReturningT())

But I'm not sure it represents even any ever written line of code as it's (for me) broken an unspecified lua code.
I would also prefer going with what you propose.

Reply:
Just to be clear, latest commit includes this proposal so there is no more getBothXXX methods, methods are returning values which are to be discarded (in legacy case, it is) or not, directly in the lua.

Reply:
Yeah, I think this is good, the chance of breaking anything is really small. And not seeing anything like it in the current scenarios.